hls4ml/model/optimizer/passes/resize_remove_constants.py

```diff
@@ -0,0 +1,35 @@
+from hls4ml.model.layers import Constant, Resize
+from hls4ml.model.optimizer import OptimizerPass
+
+
+class ResizeRemoveConstants(OptimizerPass):
+    """
+    This optimizer is intended to clean the Resize node from RoI and Scales parameters that if left cause issues in hls4ml.
+    """
+
+    def match(self, node):
+        is_match = isinstance(node, Resize) and len(node.inputs) > 1
+        return is_match
+
+    def transform(self, model, node):
+        """
+        Remove RoI and Scale Constant from new shape input.
+        """
+        # see doc here: https://onnx.ai/onnx/operators/onnx__Resize.html
+        scales_idx = 2 if len(node.inputs) == 3 or len(node.inputs) == 4 else 1
+        scales_node = node.get_input_node(node.inputs[scales_idx])
+        node.inputs[scales_idx] = ''
+        if not isinstance(scales_node, Constant):
+            raise RuntimeError("Non-constant shape inputs are not supported")
+        model.remove_node(scales_node, rewire=False)
+        if len(node.inputs) >= 2 and node.inputs[1] != '':
+            # RoI is present only if at least 2 inputs are specified
+            # RoI position is always 1 when present
+            roi_node = node.get_input_node(node.inputs[1])
+            node.inputs[1] = ''
+            if not isinstance(roi_node, Constant):
```

Comment:
Should we add a check on the constant value? What should it be if the ROI is not used?

Reply:
For the moment I considered  Resize v10 as unsupported

Reply:
Actually, the else 1 is also never true since we are not supporting v10. `scale_idx` is always 2.

Reply:
Right, RoI should be `''` or an empty vector. The first case is not supported by QONNX at the moment